File: Examples/VisualBasic/Programming-Documents/Find-Replace/ReplaceWithString.vb
Imports System.Collections
Imports System.IO
Imports System.Text.RegularExpressions
Imports Aspose.Words
Imports Aspose.Words.Fields
Imports Aspose.Words.Replacing

Class ReplaceWithString
    Public Shared Sub Run()
        ' ExStart:ReplaceWithString
        ' The path to the documents directory.
        Dim dataDir As String = RunExamples.GetDataDir_FindAndReplace()
        Dim fileName As String = "Document.doc"

        Dim doc As New Document(dataDir & fileName)
        doc.Range.Replace("sad", "bad", New FindReplaceOptions(FindReplaceDirection.Forward))

        dataDir = dataDir & Convert.ToString("ReplaceWithString_out_.doc")
        doc.Save(dataDir)
        ' ExEnd:ReplaceWithString
        Console.WriteLine(Convert.ToString(vbLf & "Text replaced with string successfully." & vbLf & "File saved at ") & dataDir)
    End Sub
End Class

File: Examples/CSharp/Programming-Documents/Find-Replace/ReplaceWithString.cs
using System.Collections;
using System.IO;
using System.Text.RegularExpressions;
using Aspose.Words;
using Aspose.Words.Fields;
using System;
using System.Drawing;
using Aspose.Words.Replacing;

namespace Aspose.Words.Examples.CSharp.Programming_Documents.Find_and_Replace
{
    class ReplaceWithString
    {
        public static void Run()
        {
            string dataDir = RunExamples.GetDataDir_FindAndReplace();

            Replace(dataDir);
            HighlightColor(dataDir);
        }

        public static void Replace(string dataDir)
        {
            // ExStart:ReplaceWithString
            // Load a Word Docx document by creating an instance of the Document class.
            Document doc = new Document();
            DocumentBuilder builder = new DocumentBuilder(doc);
            builder.Writeln("Hello _CustomerName_,");

            // Specify the search string and replace string using the Replace method.
            doc.Range.Replace("_CustomerName_", "James Bond", new FindReplaceOptions());

            // Save the result.
            doc.Save(dataDir + "Range.ReplaceSimple.docx");
            // ExEnd:ReplaceWithString
        }

        public static void HighlightColor(string dataDir)
        {
            // Load a Word Docx document by creating an instance of the Document class.
            Document doc = new Document();
            DocumentBuilder builder = new DocumentBuilder(doc);

            builder.Writeln("Hello _CustomerName_,");

            // ExStart:HighlightColor
            // Highlight word "the" with yellow color.
            FindReplaceOptions options = new FindReplaceOptions();
            options.ApplyFont.HighlightColor = Color.Yellow;

            // Replace highlighted text.
            doc.Range.Replace("Hello", "Hello", options);
            // ExEnd:HighlightColor
        }
    }

    
}
